lib/db/sprint.ts

```diff
@@ -0,0 +1,124 @@
+import prisma from "./prisma";
+
+/**
+ * Retrieves a single sprint based on provided parameters.
+ *
+ * @export
+ * @async
+ * @param {Object} params Information about the sprint.
+ * @param {number} params.id The ID of the sprint to retrieve.
+ * @param {number} params.project_id The ID of the project the sprint belongs to.
+ * @param {string} params.title The title of the sprint.
+ * @param {Date} params.start_date The start date of the sprint.
+ * @param {Date} params.end_date The end date of the sprint.
+ * @returns {Promise<Object>} The retrieved sprint object.
+ */
+export async function getSprint(params: any): Promise<Object | null> {
+  const sprint = await prisma.sprint.findFirst({
+    where: {
+      id: params.id,
+      project_id: params.project_id,
+      title: params.title,
+      start_date: params.start_date,
+      end_date: params.end_date,
+    },
+    include: { tasks: true },
+  });
+  return sprint;
+}
+
+/**
+ * Retrieves multiple sprints based on provided parameters.
+ *
+ * @export
+ * @async
+ * @param {Object} params Information about the sprints to retrieve.
+ * @param {number} params.id The ID of the sprint(s) to retrieve.
+ * @param {number} params.project_id The ID of the project the sprint(s) belong to.
+ * @param {string} params.title The title of the sprint(s).
+ * @param {Date} params.start_date The start date of the sprint(s).
+ * @param {Date} params.end_date The end date of the sprint(s).
+ * @returns {Promise<Object[]>} An array of retrieved sprint objects.
+ */
+export async function getSprints(params: any): Promise<Object[]> {
```

Comment:
For these get (search) functions, it's best practice to use partial types instead of `any`:

```javascript
getSprints(params: Partial<Sprint>): Promise<Sprint[]> {
   ...
}
```

Reply:
Additionally, thank you for creating the very descriptive doc tags! I should have given you a heads up about some of the type features in TypeScript before you got started.

Because the types prevent any users of your functions from passing incorrect parameters, you don't have to define the Sprint parameters in the doctags. Additionally, they would be hard to maintain as we make changes to the model in the future.

Reply:
Modified the doc tags. Passed in the Model Types and Custom Return Types instead.